CMakeLists.txt

```diff
@@ -83,6 +83,13 @@ IF(NOT KOKKOS_HAS_TRILINOS)
   ENDif()
   IF(NOT DEFINED ${PROJECT_NAME})
     PROJECT(Kokkos CXX)
+    IF (NOT CMAKE_BUILD_TYPE)
+      SET(DEFAULT_BUILD_TYPE "Release")
+      MESSAGE(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' as none was specified.")
+      SET(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE STRING
+          "Choose the type of build, options are: Debug, Release, RelWithDebInfo and MinSizeRel."
```

Comment:
Same question. If we only go here if build type has not been defined, we can just use a regular variable and don't need a cache variable.

Reply:
This needs to be a `CACHE` variable to have any effect.

Reply:
Touché.  This does get automatically defined in the CACHE even if not given on the command line. My bad.